.github/workflows/benchmarks.yml

```diff
@@ -1,54 +1,37 @@
 name: Benchmarks
 
 on:
+  push:
+    branches:
+      - "main"
+  pull_request:
   workflow_dispatch:
 
 jobs:
   benchmarks:
     name: Run Benchmarks
     runs-on: ubuntu-latest
-    environment: matrix-rust-bot
-    if: github.event_name == 'push'
+    strategy:
+      matrix:
+        benchmark:
+          - crypto_bench
+          - event_cache
 
     steps:
-    - name: Checkout the repo
-      uses: actions/checkout@v4
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
 
-    - name: Install Rust
-      uses: dtolnay/rust-toolchain@master
-      with:
-        toolchain: nightly-2025-06-27
-        components: rustfmt
+      - name: Setup rust toolchain, cache and cargo-codspeed binary
+        uses: moonrepo/setup-rust@ede6de059f8046a5e236c94046823e2af11ca670
+        with:
+          channel: stable
+          cache-target: release
+          bins: cargo-codspeed
 
-    - name: Run Benchmarks
-      run: cargo bench | tee benchmark-output.txt
+      - name: Build the benchmark target(s)
+        run: cargo codspeed build ${{ matrix.benchmarks }}
```

Comment:
Do we need to build with the special `profiling` profile, so as to have precise function names and so on? Not sure if this is required…

Reply:
The `bench` profile was added.